dashboard/shared/src/configurators/PrivateBuildConfigurator.ts

```diff
@@ -0,0 +1,66 @@
+import { PersistentStateManager } from "../PersistentStateManager"
+import { configureQueryProducer, DimensionConfigurator, loadDimension } from "./DimensionConfigurator"
+import { ServerConfigurator } from "./ServerConfigurator"
+import { createFilterObservable, FilterConfigurator } from "./filter"
+import { switchMap } from "rxjs"
+import { updateComponentState } from "./componentState"
+import { DataQuery, DataQueryExecutorConfiguration, DataQueryFilter } from "../dataQuery"
+
+export class PrivateBuildConfigurator extends DimensionConfigurator {
+
+  constructor() {
+    super("triggeredBy", true)
+  }
+
+  configureQuery(query: DataQuery, configuration: DataQueryExecutorConfiguration): boolean {
+    const value = this.selected.value
+    if (value == null || value.length === 0) {
+      return true
+    }
+
+    const filter: DataQueryFilter = {f: this.name, v: ""}
+    const values: Array<string> = ["", ...value]
+    configureQueryProducer(configuration, filter, values)
+    query.addFilter(filter)
+    return true
+  }
+}
+
+export function privateBuildConfigurator(serverConfigurator: ServerConfigurator,
+                                         persistentStateManager: PersistentStateManager | null,
+                                         filters: Array<FilterConfigurator> = []): DimensionConfigurator {
+  const configurator = new PrivateBuildConfigurator()
+  persistentStateManager?.add("triggeredBy", configurator.selected)
+
+  createFilterObservable(serverConfigurator, filters)
+    .pipe(
+      switchMap(() => loadDimension("triggeredBy", serverConfigurator, filters, configurator.state)),
+      updateComponentState(configurator.state),
+    )
+    .subscribe(data => {
+      if (data == null) {
+        return
+      }
+
+      configurator.values.value = data.filter((value, _n, _a) => !(value == ""))
+
+      const selectedRef = configurator.selected
```

Comment:
copypaste ;)

Reply:
It was improved a bit. I could potentially avoid `privateBuildConfigurator` completely and reuse `dimensionConfigurator` but unfortunately, I don't have a good idea how to do parametrized `configurator` creation.
Unfortunately, `function dimensionConfigurator<T extends DimensionConfigurator = DimensionConfigurator>` and later `new T` doesn't work in TypeScript since types are just "sugar" and passing instance of `configurator` looks like a bad idea for the basically builder function.